tsidp-server.go

```diff
@@ -175,7 +184,11 @@ func main() {
 		*flagEnableSTS,
 	)
 
-	srv.SetServerURL(strings.TrimSuffix(st.Self.DNSName, "."), *flagPort)
+	if *flagServerURL != "" {
+		srv.SetServerURL(*flagServerURL, *flagPort)
```

Comment:
It might be smarter to provide a list of additional `TrustedOrigins`, that way _all_ the listeners can be supported simultaneously. I'm not very familiar with OAuth2 though, so I didn't want to touch the server.go file in this PR.